packages/api/tests/endpoints/test-executions.js

```diff
@@ -147,21 +179,86 @@ test('executions can be filtered by workflow', async (t) => {
 });
 
 test('GET returns an existing execution', async (t) => {
-  const response = await request(app)
-    .get(`/executions/${fakeExecutions[0].arn}`)
+  const collectionRecord = fakeCollectionRecordFactory();
+  const asyncRecord = fakeAsyncOperationRecordFactory();
+  const parentExecutionRecord = fakeExecutionRecordFactory();
+
+  const executionPgModel = new ExecutionPgModel();
+  const collectionPgModel = new CollectionPgModel();
+  const asyncOperationsPgModel = new AsyncOperationPgModel();
+
+  const [collectionCumulusId] = await collectionPgModel.create(
+    t.context.knex,
+    collectionRecord
+  );
+
+  const [asyncOperationCumulusId] = await asyncOperationsPgModel.create(
+    t.context.knex,
+    asyncRecord
+  );
+
+  const [parentExecutionCumulusId] = await executionPgModel.create(
+    t.context.knex,
+    parentExecutionRecord
+  );
+
+  const executionRecord = await fakeExecutionRecordFactory({
+    async_operation_cumulus_id: asyncOperationCumulusId,
+    collection_cumulus_id: collectionCumulusId,
+    parent_cumulus_id: parentExecutionCumulusId,
+  });
+
+  await executionPgModel.create(
+    t.context.knex,
+    executionRecord
+  );
+  t.teardown(async () => {
+    await executionPgModel.delete(t.context.knex, executionRecord);
+    await executionPgModel.delete(t.context.knex, parentExecutionRecord);
+    await collectionPgModel.delete(t.context.knex, collectionRecord);
+    await asyncOperationsPgModel.delete(t.context.knex, asyncRecord);
+  });
+
+  const response = await request(t.context.app)
+    .get(`/executions/${executionRecord.arn}`)
+    .set('Accept', 'application/json')
+    .set('Authorization', `Bearer ${jwtAuthToken}`)
+    .expect(200);
+
+  const executionResult = response.body;
+  const expectedRecord = await translatePostgresExecutionToApiExecution(
+    executionRecord,
+    t.context.knex
+  );
+
+  t.is(executionResult.arn, executionRecord.arn);
+  t.is(executionResult.asyncOperationId, asyncRecord.id);
+  t.is(executionResult.collectionId, `${collectionRecord.name}___${collectionRecord.version}`);
+  t.is(executionResult.parentArn, parentExecutionRecord.arn);
+  t.like(executionResult, expectedRecord);
+});
+
+test('GET returns an existing execution without any foreign keys', async (t) => {
```

Comment:
Do we need this test? 🤔 

It seems like `get` will pull whatever is in the DB and not really care about foreign keys. When we create or delete an execution we definitely want to check with and without but maybe not on `get`?

Reply:
It's fair that this isn't *really* behavior on the execution *endpoint* directly, this test was written for a failure  that eventually got abstracted to logic in `translatePostgresExecutionToApiExecution`.      

If translate isn't written to not make the subquery, or with a proper join (we shouldn't do this anyway), you get either no result (bad join logic) or a failure (what happens if you remove the current branch logic in the translate method)

Reply:
Yeah, thanks for doing that! Totally makes sense.